zebra-chain/src/orchard/keys.rs

```diff
@@ -294,23 +302,31 @@ pub struct SpendValidatingKey(pub(crate) redpallas::VerificationKey<SpendAuth>);
 
 impl Eq for SpendValidatingKey {}
 
-impl From<[u8; 32]> for SpendValidatingKey {
-    fn from(bytes: [u8; 32]) -> Self {
-        Self(redpallas::VerificationKey::try_from(bytes).unwrap())
-    }
-}
-
 impl From<SpendValidatingKey> for [u8; 32] {
     fn from(ak: SpendValidatingKey) -> [u8; 32] {
         ak.0.into()
     }
 }
 
-impl From<SpendAuthorizingKey> for SpendValidatingKey {
-    fn from(ask: SpendAuthorizingKey) -> Self {
-        let sk = redpallas::SigningKey::<SpendAuth>::try_from(<[u8; 32]>::from(ask)).unwrap();
```

Comment:
This `unwrap()` will never fail, because `ask`, a scalar, is converted to a byte array and then converted back to a scalar inside `redpallas::SigningKey::<SpendAuth>::try_from`.

Which would make the fix to be: changing it to `expect()` with the reason why it's OK, and manually allowing it here if it still triggers the lint.

I realize this will require changing the entire PR (though it will be much simpler), sorry about that :sweat:

Reply:
What is the downside of leaving this as a try_from even if it will never fail ?

Reply:
It just requires more code that is not strictly required. It also makes more confusing to use, since users of the crate will have to handle the error and they won't know that the error will never be returned in the first place.

Reply:
It seems that `redpallas::SigningKey::<SpendAuth>::try_from` should be a `redpallas::SigningKey::<SpendAuth>::from` if it can't never fail. Do you think we should fix this there ?

Reply:
Anyway, maybe for another pull request. Ill add a `expect` as suggested.

Reply:
Ok, understood, thank you for the details.